packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,707 @@
+        Name: this.props.jobName,
+        ExecutionRoleArn: this.role.roleArn,
+        ReleaseLabel: this.props.releaseLabel.label,
+        JobDriver: {
+          SparkSubmitJobDriver: {
+            EntryPoint: this.props.jobDriver?.sparkSubmitJobDriver?.entryPoint.value,
+            EntryPointArguments: this.props.jobDriver?.sparkSubmitJobDriver?.entryPointArguments?.value,
+            SparkSubmitParameters: this.props.jobDriver?.sparkSubmitJobDriver?.sparkSubmitParameters,
+          },
+        },
+        ConfigurationOverrides: {
+          ApplicationConfiguration: cdk.listMapper(this.ApplicationConfigPropertyToJson)(this.props.applicationConfig),
+          MonitoringConfiguration: {
+            CloudWatchMonitoringConfiguration: {
+              LogGroupName: (this.props.monitoring?.logging === true)
+                ? this.renderLogGroupName() // automatically generated name https://docs.aws.amazon.com/cdk/api/latest/typescript/api/aws-logs/loggroup.html#aws_logs_LogGroup_synopsis
+                : this.props.monitoring?.logGroup?.logGroupName,
+              LogStreamNamePrefix: this.props.monitoring?.logStreamNamePrefix,
+            },
+            PersistentAppUI: (this.props.monitoring?.persistentAppUI === false)
+              ? 'DISABLED'
+              : 'ENABLED',
+            S3MonitoringConfiguration: {
+              LogUri: (this.props.monitoring?.logging === true)
+                ? `s3://${this.renderS3BucketName()}`// automatically generated name https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-s3.Bucket.html#bucketname
+                : `s3://${this.props.monitoring?.logBucket?.bucketName}`,
+            },
+          },
+          Tags: this.props.tags ? this.renderTags(this.props.tags) : undefined,
+        },
+      }),
+    };
+  }
+
+
+  /**
+   * Render the EMR Containers ConfigurationProperty as JSON
+   *
+   * @param property
+   */
+  private applicationConfigPropertyToJson(property: ApplicationConfiguration) {
+    return {
+      Classification: cdk.stringToCloudFormation(property.classification.classificationStatement),
+      Properties: cdk.objectToCloudFormation(property.properties),
+      Configurations: cdk.listMapper(this.ApplicationConfigPropertyToJson)(property.nestedConfig),
+    };
+  }
+
+  private renderTags(tags?: { [key: string]: any }): { Key: string, Value: any } {
+    return tags ? { Tags: Object.keys(tags).map((key) => ({ Key: key, Value: tags[key] })) } : {};
+  }
+
+  /**
+   * Creates a Cloudwatch log group
+   *
+   * @returns name if user enables props.monitoring.logging to be true
+   **/
+  private renderLogGroupName(): string {
+    return new logs.LogGroup(this, 'Emr Containers Default Cloudwatch Log Group').logGroupName;
+  }
+
+  /**
+   * Creates a S3 bucket
+   *
+   * @returns an automatically generated bucket name
+   **/
+  private renderS3BucketName(): string {
+    return new s3.Bucket(this, 'Emr Containers Default S3 Bucket').bucketName;
```

Comment:
Will this be a randomly generated S3 bucket name? Same with the log group name above.

Reply:
Yes this will be randomly generated from the uniqueId.